pkg/policy/api/selector.go

```diff
@@ -159,13 +187,29 @@ func NewESFromLabels(lbls ...*labels.Label) EndpointSelector {
 // NewESFromMatchRequirements creates a new endpoint selector from the given
 // match specifications: An optional set of labels that must match, and
 // an optional slice of LabelSelectorRequirements.
+//
+// If the caller intends to reuse 'matchLabels' or 'reqs' after creating the
+// EndpointSelector, they must make a copy of the parameter.
 func NewESFromMatchRequirements(matchLabels map[string]string, reqs []metav1.LabelSelectorRequirement) EndpointSelector {
 	labelSelector := &metav1.LabelSelector{
 		MatchLabels:      matchLabels,
 		MatchExpressions: reqs,
 	}
 	return EndpointSelector{
 		LabelSelector: labelSelector,
+		requirements:  labelSelectorToRequirements(labelSelector),
+	}
+}
+
+// SyncRequirementsWithLabelSelector returns an EndpointSelector which ensures
+// that the requirements within the specified EndpointSelector are in sync
+// with the LabelSelector. This is because the LabelSelector has publicly
+// accessible fields, which can be updated without concurrently updating the
+// requirements, so the two fields can become out of sync.
+func (n *EndpointSelector) SyncRequirementsWithLabelSelector() EndpointSelector {
+	return EndpointSelector{
```

Comment:
Why don't you update the requirements for the selector, in place?

Reply:
Fixed in latest upload.